src/systems/optimal_control_interface.jl

```diff
@@ -467,24 +552,33 @@ function prepare_and_optimize! end
 function get_t_values end
 function get_U_values end
 function get_V_values end
+function get_P_values end
 function successful_solve end
 
 """
     solve(prob::AbstractDynamicOptProblem, solver::AbstractCollocation; verbose = false, kwargs...)
 
 - kwargs are used for other options. For example, the `plugin_options` and `solver_options` will propagated to the Opti object in CasADi.
 """
-function DiffEqBase.solve(prob::AbstractDynamicOptProblem,
+function DiffEqBase.solve(prob::SciMLBase.AbstractDynamicOptProblem,
```

Comment:
Could you also change the method to `CommonSolve.solve`? I didn't notice the transitive dependency and it would be nice to fix.

```suggestion
function CommonSolve.solve(prob::SciMLBase.AbstractDynamicOptProblem,
```